storage/postgres/broker.go

```diff
@@ -17,37 +17,170 @@
 package postgres
 
 import (
-	"github.com/Peripli/service-manager/types"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/Peripli/service-manager/rest"
+	"github.com/Peripli/service-manager/storage"
 	"github.com/jmoiron/sqlx"
-	"context"
+	"github.com/lib/pq"
+	"github.com/sirupsen/logrus"
 )
 
 type brokerStorage struct {
 	db *sqlx.DB
 }
 
-func (storage *brokerStorage) Create(ctx context.Context, broker *types.Broker) error {
-	return nil
+type fetcher func(dest interface{}, query string, args ...interface{}) error
+
+func (store *brokerStorage) Create(broker *rest.Broker) error {
```

Comment:
the storage api provides a common interface to the persistency layer so it has to use a common model, currently this is the rest model
Note, we don't aim for the storage api ro be reusable in other contexts

Reply:
Well, the storage needs to return everything from the database and doesn't need to know about how everyone else will use this data. For example, this breaks the "call service broker" functionality written here https://github.com/Peripli/service-manager/pull/46/files#diff-404f10463ebdd7100ab15d74f6763e45R197 as the credentials are not returned by the storage because it is concerned with another layer.

Reply:
yeah, I think we discussed this when we talked about eager / lazy fetching of related tables - this here should be doing lazy fetching probably